pkg/controller/plan/kubevirt.go

```diff
@@ -1166,6 +1181,97 @@ func (r *KubeVirt) virtualMachine(vm *plan.VMStatus) (object *cnv.VirtualMachine
 	return
 }
 
+// Attempt to find a suitable preference.
+func (r *KubeVirt) vmPreference(vm *plan.VMStatus) (virtualMachine *cnv.VirtualMachine, err error) {
+	config, err := r.getOsMapConfig(r.Source.Provider.Type())
+	if err != nil {
+		return
+	}
+	preferenceName, err := r.Builder.PreferenceName(vm.Ref, config)
+	if err != nil {
+		return
+	}
+
+	preferenceName, kind, err := r.getPreference(vm, preferenceName)
+	if err != nil {
+		return
+	}
+
+	virtualMachine = r.emptyVm(vm)
+	virtualMachine.Spec.Preference = &cnv.PreferenceMatcher{Name: preferenceName, Kind: kind}
+	return
+}
+
+func (r *KubeVirt) getOsMapConfig(providerType v1beta1.ProviderType) (configMap *core.ConfigMap, err error) {
+	cm := &core.ConfigMap{}
+	var configMapName string
+	switch providerType {
+	case api.VSphere:
+		configMapName = vsphereOsConfigMap
+	case api.OVirt:
+		configMapName = ovirtOsConfigMap
+	default:
+		return
+	}
+	err = r.Client.Get(
+		context.TODO(),
+		client.ObjectKey{Name: configMapName},
```

Comment:
now the problem with get is to know the correct namespace :)

Reply:
If the default ServiceAccount is present in the Pod, you can get the namespace the Pod is running in by reading `/var/run/secrets/kubernetes.io/serviceaccount/namespace`. This necessitates that the ConfigMap is located in the same namespace as the Pod however.

Reply:
we have envVar holding it within the pod, i used this approach.